kedro/runner/sequential_runner.py

```diff
@@ -46,11 +46,16 @@ def __init__(
             is_async=is_async, extra_dataset_patterns=self._extra_dataset_patterns
         )
 
+    def _get_executor(self, max_workers: int) -> ThreadPoolExecutor:
```

Comment:
Is this necessary to create a thread for SeqeuntialRunner?

Reply:
Strictly speaking it's not necessary, because it's just one thread. I did it to make it more consistent with the other runners and to make it possible to have the shared logic in the abstract `_run()` method:
```
        with self._get_executor(max_workers) as pool:
            while True:
```

Reply:
Would that have any consequences for non thread-safe datasets?

Reply:
For `SequentialRunner` it will only use 1 thread, so it shouldn't have any consequences (although I'm sure there are edge cases). A quick search in our datasets doesn't show any mentions of thread-safety. Do you know of any specific dataset not being thread-safe?

Reply:
I don't have any dataset in mind, was just curious if we have considered this. If we don't have any indication that this doesn't make a difference in any way, we should just go ahead and deal with it only if it causes problems later.